vyper/ast/folding.py

```diff
@@ -170,6 +172,15 @@ def replace_user_defined_constants(vyper_module: vy_ast.Module) -> int:
     """
     changed_nodes = 0
 
+    # manually populate namespace with structs
+    namespace = get_namespace()
```

Comment:
I did not override the namespace here since no namespace has been created yet in the folding phase.

Reply:
i'm not sure this is the best approach, but in any case, you would want to use `with namespace.enter_scope():` here

Reply:
I have made the change. However, the namespace still needs to be manually cleared to prevent subsequent namespace collision since it is the first instantiation of namespace.